File: Nx3CS/bin/test/src/nx3/NBar.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class NBar : global::haxe.lang.HxObject {
		public    NBar(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    NBar(global::Array<object> parts, global::nx3.EBarType type, global::nx3.ETime time, global::nx3.EDisplayALN timeDisplay, global::nx3.EAllotment allotment, global::haxe.lang.Null<double> spacing){
			unchecked {
				#line 13 "F:\\nx3\\src\\nx3\\NBar.hx"
				global::nx3.NBar.__hx_ctor_nx3_NBar(this, parts, type, time, timeDisplay, allotment, spacing);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_NBar(global::nx3.NBar __temp_me67, global::Array<object> parts, global::nx3.EBarType type, global::nx3.ETime time, global::nx3.EDisplayALN timeDisplay, global::nx3.EAllotment allotment, global::haxe.lang.Null<double> spacing){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\NBar.hx"
				global::haxe.lang.Null<double> __temp_spacing66 = ( ( ! (spacing.hasValue) ) ? (new global::haxe.lang.Null<double>(((double) (8) ), true)) : (((global::haxe.lang.Null<double>) (spacing) )) );
				__temp_me67.nparts = parts;
				{
					#line 16 "F:\\nx3\\src\\nx3\\NBar.hx"
					int _g = 0;
					#line 16 "F:\\nx3\\src\\nx3\\NBar.hx"
					while (( _g < parts.length )){
						#line 16 "F:\\nx3\\src\\nx3\\NBar.hx"
						global::nx3.NPart part = ((global::nx3.NPart) (parts[_g]) );
						#line 16 "F:\\nx3\\src\\nx3\\NBar.hx"
						 ++ _g;
						#line 16 "F:\\nx3\\src\\nx3\\NBar.hx"
						part.nbar = __temp_me67;
					}
					
				}
				
				#line 18 "F:\\nx3\\src\\nx3\\NBar.hx"
				if (( type == default(global::nx3.EBarType) )) {
					#line 18 "F:\\nx3\\src\\nx3\\NBar.hx"
					__temp_me67.type = global::nx3.EBarType.Normal;
				}
				 else {
					#line 18 "F:\\nx3\\src\\nx3\\NBar.hx"
					__temp_me67.type = type;
				}
				
				#line 20 "F:\\nx3\\src\\nx3\\NBar.hx"
				__temp_me67.time = time;
				if (( timeDisplay == default(global::nx3.EDisplayALN) )) {
					#line 21 "F:\\nx3\\src\\nx3\\NBar.hx"
					__temp_me67.timeDisplay = global::nx3.EDisplayALN.Layout;
				}
				 else {
					#line 21 "F:\\nx3\\src\\nx3\\NBar.hx"
					__temp_me67.timeDisplay = timeDisplay;
				}
				
				#line 22 "F:\\nx3\\src\\nx3\\NBar.hx"
				if (( allotment == default(global::nx3.EAllotment) )) {
					#line 22 "F:\\nx3\\src\\nx3\\NBar.hx"
					__temp_me67.allotment = global::nx3.EAllotment.Logaritmic;
				}
				 else {
					#line 22 "F:\\nx3\\src\\nx3\\NBar.hx"
					__temp_me67.allotment = allotment;
				}
				
				#line 23 "F:\\nx3\\src\\nx3\\NBar.hx"
				__temp_me67.spacing = __temp_spacing66.@value;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				return new global::nx3.NBar(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				return new global::nx3.NBar(((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (arr[0]) ))) ), ((global::nx3.EBarType) (arr[1]) ), ((global::nx3.ETime) (arr[2]) ), ((global::nx3.EDisplayALN) (arr[3]) ), ((global::nx3.EAllotment) (arr[4]) ), global::haxe.lang.Null<object>.ofDynamic<double>(arr[5]));
			}
			#line default
		}
		
		
		public  global::nx3.NScore nscore;
		
		public  global::Array<object> nparts;
		
		public  global::nx3.EBarType type;
		
		public  global::nx3.ETime time;
		
		public  global::nx3.EDisplayALN timeDisplay;
		
		public  global::nx3.EAllotment allotment;
		
		public  double spacing;
		
		public virtual   global::nx3.NNote getNNote(int partIdx, int voiceIdx, int noteIdx){
			unchecked {
				#line 35 "F:\\nx3\\src\\nx3\\NBar.hx"
				return this.getNPart(partIdx).getNVoice(voiceIdx).getNNote(noteIdx);
			}
			#line default
		}
		
		
		public virtual   global::nx3.NPart getNPart(int idx){
			unchecked {
				#line 37 "F:\\nx3\\src\\nx3\\NBar.hx"
				if (( ( idx < 0 ) || ( idx > this.nparts.length ) )) {
					#line 37 "F:\\nx3\\src\\nx3\\NBar.hx"
					return default(global::nx3.NPart);
				}
				 else {
					#line 37 "F:\\nx3\\src\\nx3\\NBar.hx"
					return ((global::nx3.NPart) (this.nparts[idx]) );
				}
				
			}
			#line default
		}
		
		
		public virtual   object iterator(){
			unchecked {
				#line 39 "F:\\nx3\\src\\nx3\\NBar.hx"
				return ((object) (new global::_Array.ArrayIterator<object>(((global::Array<object>) (this.nparts) ))) );
			}
			#line default
		}
		
		
		public  int length;
		
		public virtual   int get_length(){
			unchecked {
				#line 41 "F:\\nx3\\src\\nx3\\NBar.hx"
				return this.nparts.length;
			}
			#line default
		}
		
		
		public override   double __hx_setField_f(string field, int hash, double @value, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				switch (hash){
					case 520590566:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.length = ((int) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					case 297481091:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.spacing = ((double) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return base.__hx_setField_f(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				switch (hash){
					case 520590566:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.length = ((int) (global::haxe.lang.Runtime.toInt(@value)) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					case 297481091:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.spacing = ((double) (global::haxe.lang.Runtime.toDouble(@value)) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					case 239117604:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.allotment = ((global::nx3.EAllotment) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					case 765448629:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.timeDisplay = ((global::nx3.EDisplayALN) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					case 1291635725:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.time = ((global::nx3.ETime) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					case 1292432058:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.type = ((global::nx3.EBarType) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					case 949436914:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.nparts = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					case 1947935780:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						this.nscore = ((global::nx3.NScore) (@value) );
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return @value;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				switch (hash){
					case 261031087:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("get_length") ), ((int) (261031087) ))) );
					}
					
					
					case 520590566:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						if (handleProperties) {
							#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
							return this.get_length();
						}
						 else {
							#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
							return this.length;
						}
						
					}
					
					
					case 328878574:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("iterator") ), ((int) (328878574) ))) );
					}
					
					
					case 1796599563:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getNPart") ), ((int) (1796599563) ))) );
					}
					
					
					case 1775117066:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getNNote") ), ((int) (1775117066) ))) );
					}
					
					
					case 297481091:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.spacing;
					}
					
					
					case 239117604:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.allotment;
					}
					
					
					case 765448629:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.timeDisplay;
					}
					
					
					case 1291635725:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.time;
					}
					
					
					case 1292432058:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.type;
					}
					
					
					case 949436914:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.nparts;
					}
					
					
					case 1947935780:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.nscore;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_getField_f(string field, int hash, bool throwErrors, bool handleProperties){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				switch (hash){
					case 520590566:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						if (handleProperties) {
							#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
							return ((double) (this.get_length()) );
						}
						 else {
							#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
							return ((double) (this.length) );
						}
						
					}
					
					
					case 297481091:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.spacing;
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return base.__hx_getField_f(field, hash, throwErrors, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				switch (hash){
					case 261031087:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.get_length();
					}
					
					
					case 328878574:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.iterator();
					}
					
					
					case 1796599563:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.getNPart(((int) (global::haxe.lang.Runtime.toInt(dynargs[0])) ));
					}
					
					
					case 1775117066:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return this.getNNote(((int) (global::haxe.lang.Runtime.toInt(dynargs[0])) ), ((int) (global::haxe.lang.Runtime.toInt(dynargs[1])) ), ((int) (global::haxe.lang.Runtime.toInt(dynargs[2])) ));
					}
					
					
					default:
					{
						#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				baseArr.push("length");
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				baseArr.push("spacing");
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				baseArr.push("allotment");
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				baseArr.push("timeDisplay");
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				baseArr.push("time");
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				baseArr.push("type");
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				baseArr.push("nparts");
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				baseArr.push("nscore");
				#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
				{
					#line 10 "F:\\nx3\\src\\nx3\\NBar.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/NBar.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class NBar extends haxe.lang.HxObject
{
	public    NBar(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    NBar(haxe.root.Array<nx3.NPart> parts, nx3.EBarType type, nx3.ETime time, nx3.EDisplayALN timeDisplay, nx3.EAllotment allotment, java.lang.Object spacing)
	{
		nx3.NBar.__hx_ctor_nx3_NBar(this, parts, type, time, timeDisplay, allotment, spacing);
	}
	
	
	public static   void __hx_ctor_nx3_NBar(nx3.NBar __temp_me60, haxe.root.Array<nx3.NPart> parts, nx3.EBarType type, nx3.ETime time, nx3.EDisplayALN timeDisplay, nx3.EAllotment allotment, java.lang.Object spacing)
	{
		java.lang.Object __temp_spacing59 = ( (( spacing == null )) ? (((java.lang.Object) (8) )) : (((java.lang.Object) (spacing) )) );
		__temp_me60.nparts = parts;
		{
			int _g = 0;
			while (( _g < parts.length ))
			{
				nx3.NPart part = parts.__get(_g);
				 ++ _g;
				part.nbar = __temp_me60;
			}
			
		}
		
		if (( type == null )) 
		{
			__temp_me60.type = nx3.EBarType.Normal;
		}
		 else 
		{
			__temp_me60.type = type;
		}
		
		__temp_me60.time = time;
		if (( timeDisplay == null )) 
		{
			__temp_me60.timeDisplay = nx3.EDisplayALN.Layout;
		}
		 else 
		{
			__temp_me60.timeDisplay = timeDisplay;
		}
		
		if (( allotment == null )) 
		{
			__temp_me60.allotment = nx3.EAllotment.Logaritmic;
		}
		 else 
		{
			__temp_me60.allotment = allotment;
		}
		
		__temp_me60.spacing = ((double) (haxe.lang.Runtime.toDouble(__temp_spacing59)) );
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.NBar(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.NBar(((haxe.root.Array<nx3.NPart>) (arr.__get(0)) ), ((nx3.EBarType) (arr.__get(1)) ), ((nx3.ETime) (arr.__get(2)) ), ((nx3.EDisplayALN) (arr.__get(3)) ), ((nx3.EAllotment) (arr.__get(4)) ), ((java.lang.Object) (arr.__get(5)) ));
	}
	
	
	public  nx3.NScore nscore;
	
	public  haxe.root.Array<nx3.NPart> nparts;
	
	public  nx3.EBarType type;
	
	public  nx3.ETime time;
	
	public  nx3.EDisplayALN timeDisplay;
	
	public  nx3.EAllotment allotment;
	
	public  double spacing;
	
	public   nx3.NNote getNNote(int partIdx, int voiceIdx, int noteIdx)
	{
		return this.getNPart(partIdx).getNVoice(voiceIdx).getNNote(noteIdx);
	}
	
	
	public   nx3.NPart getNPart(int idx)
	{
		if (( ( idx < 0 ) || ( idx > this.nparts.length ) )) 
		{
			return null;
		}
		 else 
		{
			return this.nparts.__get(idx);
		}
		
	}
	
	
	public   java.lang.Object iterator()
	{
		return ((java.lang.Object) (new _Array.ArrayIterator<nx3.NPart>(((haxe.root.Array<nx3.NPart>) (this.nparts) ))) );
	}
	
	
	public  int length;
	
	public   int get_length()
	{
		return this.nparts.length;
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef405 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef405 = false;
						this.length = ((int) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -2012158909:
				{
					if (field.equals("spacing")) 
					{
						__temp_executeDef405 = false;
						this.spacing = ((double) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef405) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef406 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef406 = false;
						this.length = ((int) (haxe.lang.Runtime.toInt(value)) );
						return value;
					}
					
					break;
				}
				
				
				case -1036496156:
				{
					if (field.equals("nscore")) 
					{
						__temp_executeDef406 = false;
						this.nscore = ((nx3.NScore) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -2012158909:
				{
					if (field.equals("spacing")) 
					{
						__temp_executeDef406 = false;
						this.spacing = ((double) (haxe.lang.Runtime.toDouble(value)) );
						return value;
					}
					
					break;
				}
				
				
				case -1039323342:
				{
					if (field.equals("nparts")) 
					{
						__temp_executeDef406 = false;
						this.nparts = ((haxe.root.Array<nx3.NPart>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 369993124:
				{
					if (field.equals("allotment")) 
					{
						__temp_executeDef406 = false;
						this.allotment = ((nx3.EAllotment) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3575610:
				{
					if (field.equals("type")) 
					{
						__temp_executeDef406 = false;
						this.type = ((nx3.EBarType) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -1919414987:
				{
					if (field.equals("timeDisplay")) 
					{
						__temp_executeDef406 = false;
						this.timeDisplay = ((nx3.EDisplayALN) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3560141:
				{
					if (field.equals("time")) 
					{
						__temp_executeDef406 = false;
						this.time = ((nx3.ETime) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef406) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef407 = true;
			switch (field.hashCode())
			{
				case 974314479:
				{
					if (field.equals("get_length")) 
					{
						__temp_executeDef407 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("get_length"))) );
					}
					
					break;
				}
				
				
				case -1036496156:
				{
					if (field.equals("nscore")) 
					{
						__temp_executeDef407 = false;
						return this.nscore;
					}
					
					break;
				}
				
				
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef407 = false;
						if (handleProperties) 
						{
							return this.get_length();
						}
						 else 
						{
							return this.length;
						}
						
					}
					
					break;
				}
				
				
				case -1039323342:
				{
					if (field.equals("nparts")) 
					{
						__temp_executeDef407 = false;
						return this.nparts;
					}
					
					break;
				}
				
				
				case 1182533742:
				{
					if (field.equals("iterator")) 
					{
						__temp_executeDef407 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("iterator"))) );
					}
					
					break;
				}
				
				
				case 3575610:
				{
					if (field.equals("type")) 
					{
						__temp_executeDef407 = false;
						return this.type;
					}
					
					break;
				}
				
				
				case 1959892939:
				{
					if (field.equals("getNPart")) 
					{
						__temp_executeDef407 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getNPart"))) );
					}
					
					break;
				}
				
				
				case 3560141:
				{
					if (field.equals("time")) 
					{
						__temp_executeDef407 = false;
						return this.time;
					}
					
					break;
				}
				
				
				case 1959846858:
				{
					if (field.equals("getNNote")) 
					{
						__temp_executeDef407 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getNNote"))) );
					}
					
					break;
				}
				
				
				case -1919414987:
				{
					if (field.equals("timeDisplay")) 
					{
						__temp_executeDef407 = false;
						return this.timeDisplay;
					}
					
					break;
				}
				
				
				case -2012158909:
				{
					if (field.equals("spacing")) 
					{
						__temp_executeDef407 = false;
						return this.spacing;
					}
					
					break;
				}
				
				
				case 369993124:
				{
					if (field.equals("allotment")) 
					{
						__temp_executeDef407 = false;
						return this.allotment;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef407) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef408 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef408 = false;
						if (handleProperties) 
						{
							return ((double) (this.get_length()) );
						}
						 else 
						{
							return ((double) (this.length) );
						}
						
					}
					
					break;
				}
				
				
				case -2012158909:
				{
					if (field.equals("spacing")) 
					{
						__temp_executeDef408 = false;
						return this.spacing;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef408) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef409 = true;
			switch (field.hashCode())
			{
				case 974314479:
				{
					if (field.equals("get_length")) 
					{
						__temp_executeDef409 = false;
						return this.get_length();
					}
					
					break;
				}
				
				
				case 1959846858:
				{
					if (field.equals("getNNote")) 
					{
						__temp_executeDef409 = false;
						return this.getNNote(((int) (haxe.lang.Runtime.toInt(dynargs.__get(0))) ), ((int) (haxe.lang.Runtime.toInt(dynargs.__get(1))) ), ((int) (haxe.lang.Runtime.toInt(dynargs.__get(2))) ));
					}
					
					break;
				}
				
				
				case 1182533742:
				{
					if (field.equals("iterator")) 
					{
						__temp_executeDef409 = false;
						return this.iterator();
					}
					
					break;
				}
				
				
				case 1959892939:
				{
					if (field.equals("getNPart")) 
					{
						__temp_executeDef409 = false;
						return this.getNPart(((int) (haxe.lang.Runtime.toInt(dynargs.__get(0))) ));
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef409) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("length");
		baseArr.push("spacing");
		baseArr.push("allotment");
		baseArr.push("timeDisplay");
		baseArr.push("time");
		baseArr.push("type");
		baseArr.push("nparts");
		baseArr.push("nscore");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


